cmd/goprotoc/codegen.go

```diff
@@ -107,10 +107,27 @@ func doCodeGen(outputs map[string]string, fds []*desc.FileDescriptor, pluginDefs
 		}
 		w, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
 		if err != nil {
+			if os.IsNotExist(err) {
+				// The file does not have to exist since we pass o.O_CREATE but the above
+				// open can return error if the directory does not exist, and protoc
+				// makes all necessary subdirectories as part of generation.
+				if err := os.MkdirAll(filepath.Dir(fileName), 0755); err != nil {
+					return err
+				}
+				// Try again after making the directory.
+				w, err = os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+		if _, err := io.Copy(w, fileContents); err != nil {
+			_ = w.Close()
```

Comment:
Doh! Good catch. That makes me glad your ulimit was so low to make this an easier find.

Reply:
Yea it's been causing me to find bugs for the last two days :-)